### ISSUE TITLE
Remove side-effect Terminology From ZIO Docs (API and Website)

### ISSUE THREAD

**Issue by khajavi** [zio/zio#7837]
Following [this conversation](https://github.com/zio/zio/pull/7834#discussion_r1107365405), we need to remove side-effect terminology from the documentation.

**Comment by hearnadam** [comment]
@khajavi is this done?

**Comment by khajavi** [comment]
@hearnadam No, please keep it open! It requires a notable amount of work to be done!
